### DISCUSSION TITLE
React-hook-form: Error messages not displayed after form submission. I need to reload the page manually

### DISCUSSION THREAD

**Discussion by marijana-rukavina** [react-hook-form/react-hook-form#10327] [Q&A] [answered]
I am using react-hook-form and yup to validate my form field input.  
I also use the `ErrorMessage` component from react-hook-form to display multiple messages.  
I have an issue: when I submit the form, and after that immediately enter invalid input in the field, then the error message is not displayed.  
However, my submit button is disabled which means the validation kicks in but the message is not displaying.  
After I reload the page manually, then everything starts working as expected - the error messages are displayed again.  

Here are the snippets from my code:  

    import React, { useState } from "react";
    import { useForm } from "react-hook-form";
    import { yupResolver } from "@hookform/resolvers/yup";
    import { ErrorMessage } from "@hookform/error-message";
    import validationSchema from "../validations/validationSchema";
    
    export default function Report(props) {
      const {
        register,
        handleSubmit,
        formState: { errors, isValid },
      } = useForm({
        mode: "onBlur",
        // reValidateMode: "onChange",
        criteriaMode: "all",
        resolver: yupResolver(validationSchema),
        defaultValues: {
          closedBugs: "",
        },
      });
    
      const [state, setState] = useState({
        closedBugs: "",
        reportedIncidents: "",
        reportedBugs: "",
      });
    
      const onFormSubmit = (data, event) => {
        event.preventDefault();
        props.onSubmit(state);
      };
        setState({
          ...state,
          [key]: value,
        });
      };
    
      const modifyBugsValues = (value) => {
        if (value === "") {
          // needed in order for yup to return required field validation in case of empty field
          return undefined;
        } else {
          // needed in order for yup to treat 1,2 as 1.2 and return integer validation
          value = value.replace(/,/g, ".");
          return value;
        }
      };
    
      return (
        <div className="container">
          <div className="ui segment report">
            <form onSubmit={handleSubmit(onFormSubmit)} className="ui form">
              <h4 className="ui dividing header">Project information</h4>
              <h4 className="ui dividing header">Bug metrics</h4>
              <div className="three fields">
                <div className="field mandatory-metric">
                  <label>closed bugs during month</label>
                  <div className="field">
                    <input
                      type="text"
                      {...register("closedBugs", {
                        setValueAs: modifyBugsValues,
                      })}
                      onChange={(e) => handleChange(e.target.name, e.target.value)}
                      name="closedBugs"
                      placeholder="1"
                      required
                    />
    
                    <ErrorMessage
                      errors={errors}
                      name="closedBugs"
                      render={({ messages }) => {
                        console.log("Messages" + JSON.stringify(messages));
                        return (
                          messages &&
                          Object.entries(messages).map(([type, message]) => (
                            <p className="error" key={type}>
                              {message}
                            </p>
                          ))
                        );
                      }}
                    />
                  </div>
                </div>
              <button
                type="submit"
                className="ui secondary large button"
                data-cy="submitBtn"
                disabled={!isValid}
                tabIndex={0}
              >
                Submit report
              </button>
            </form>
          </div>
        </div>
      );
    }

**Comment by leapful** [comment]
@marijana-rukavina 

Please provide a codesandbox url to re-produce the issue.

**Comment by marijana-rukavina** [comment]
@leapful 
Thanks for your reply.
I never used codesandbox before. So please understand if I didn't get something right :) 
This is the link, I hope you can access it. 
https://codesandbox.io/s/interesting-golick-zkmddp
The scenario to reproduce:

- leave the field empty =Error message appears and submit button is disabled.
- enter number 12
- click Submit Report button
- click No
- empty the field again => Error message doesn't appear but the submit button is disabled

**Comment by leapful** [comment, marked as answer]
@marijana-rukavina 

You are using `mode=onBlur` but `reValidationMode=onChange` therefore it has been conflict between validation and re-validation stategy.

You can apply `reValidationMode=onBlur` to solve the problem.

Working demo: https://codesandbox.io/s/elegant-kowalevski-h2jnsg?file=/src/Report.js

**Comment by marijana-rukavina** [comment]
Thanks, that solved my issue. :)
